### PR TITLE
algebra cleanup in FY5253 and Easter offsets

### PR DESCRIPTION
**Pull request by jbrockmendel**
Same types of cleanups you've gotten used to seeing elsewhere, now applied to FY5253 and Easter.  This should be orthogonal to others.

### CODE REVIEW

File: pandas/tseries/offsets.py [hunk 1]
```diff
@@ -2013,63 +1993,45 @@ def apply(self, other):
...
-            elif other < prev_year:
-                year = other.year - 1
-                n -= 1
+        if other == prev_year:
+            n -= 1
+        elif other == cur_year:
+            pass
+        elif other == next_year:
+            n += 1
+            # TODO: Not hit in tests
```
Comment (jreback) [inline pandas/tseries/offsets.py]: can we hit these in tests

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tseries/offsets.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tseries/offsets.py]: It wasn't obvious to me how to hit it, so I left the note to try again next time around.  (which will be soon.  We're far from done with offsets PRs)

**Comment by jbrockmendel** [comment]
I think there's one more algebra/cleanup after this, then we're jumping into bugfixes.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
I doubt we have very many tests for this. @gfyoung can you have a look and see if the changes look ok.

**Comment by gfyoung** [comment]
@jreback : I'll take a look later this evening.  If all looks, can I just merge this?

**Comment by jreback** [comment]
yep

File: pandas/tseries/offsets.py (hunk 1 above)
Comment (gfyoung) [reply pandas/tseries/offsets.py]: @jbrockmendel : If you don't mind explaining, how do you know that this particular branch isn't hit?  What about the other branches?

Comment (gfyoung) [reply pandas/tseries/offsets.py]: BTW, the changes themselves look **OK** but would like to get some more color on the questions I just posed to you before merging.

Comment (jbrockmendel) [reply pandas/tseries/offsets.py]: Put in a bunch of “assert false” statements as a quick coverage check.

Comment (gfyoung) [reply pandas/tseries/offsets.py]: Ah, I see.  From which test files did those `assert false` failures then originate?

Comment (jbrockmendel) [reply pandas/tseries/offsets.py]: Almost certainly `tests.tseries.offsets.test_offsets`

**Comment by jbrockmendel** [comment]
#14774 has a use case that hits one of the existing `assert False` statements (i.e. not one of the ones discussed above for coverage checks).  I'll take a look at that after this gets closed.

File: pandas/tseries/offsets.py (hunk 1 above)
Comment (gfyoung) [reply pandas/tseries/offsets.py]: Okay, sounds good.  Have a look at those tests once I merge to see how we can increase coverage.

**Comment by gfyoung** [comment]
@jbrockmendel : Merged and closed.  Thanks!

**Pull request merged**